### ISSUE TITLE
CVodeF returned "CV_RHSFUNC_FAIL"

### ISSUE THREAD

**Issue by juliuskittler** [casadi/casadi#2661]
Dear casadi team, 

New to Casadi, I am encountering an issue where my optimization gets stuck after a certain number of iterations (always at the same iteration if I repeat the optimization). I am using the `nlpsol` optimizer with the following set up:

```
opts = {'eval_errors_fatal': False, 'verbose_init': False, 'error_on_fail': False, 'ipopt': {'fixed_variable_treatment': 'make_constraint'}}
prob = {'f': J, 'x': casadi.vertcat(var_1, var_2, var_3), 'g': casadi.vertcat(*g)}
solver = casadi.nlpsol('solver', 'ipopt', prob, opts)
```

After iteration 188, the minimization just stops (see bottom of the printout below). Among others, the error says `CVodeF returned "CV_RHSFUNC_FAIL"`. 

_Any hints regarding possible causes (and solutions) of this issue would be greatly appreciated._ Unfortunately, I don't have permission to share my full code example.

```This is Ipopt version 3.12.3, running with linear solver mumps.
NOTE: Other linear solvers might be more efficient (see Ipopt documentation).

Number of nonzeros in equality constraint Jacobian...:        0
Number of nonzeros in inequality constraint Jacobian.:        9
Number of nonzeros in Lagrangian Hessian.............:       21

Total number of variables............................:        6
                     variables with only lower bounds:        0
                variables with lower and upper bounds:        6
                     variables with only upper bounds:        0
Total number of equality constraints.................:        0
Total number of inequality constraints...............:        2
        inequality constraints with only lower bounds:        1
   inequality constraints with lower and upper bounds:        1
        inequality constraints with only upper bounds:        0

iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls
   0  2.4467991e+02 4.91e+00 3.36e+01  -1.0 0.00e+00    -  0.00e+00 0.00e+00   0
   1  2.4467373e+02 4.91e+00 8.98e+01  -1.0 4.31e+00   4.0 2.15e-02 2.30e-04h  1
   2  2.4447703e+02 4.91e+00 8.84e+02  -1.0 1.36e+01   3.5 7.34e-03 8.45e-04h  1
   3  2.4130129e+02 4.83e+00 1.00e+03  -1.0 1.16e+01   3.0 2.44e-03 3.79e-02f  2
   4  2.3944838e+02 4.74e+00 7.40e+02  -1.0 8.54e+00   3.5 1.90e-02 6.93e-03f  3
   5  2.3454323e+02 4.25e+00 1.83e+03  -1.0 6.94e-01   3.9 2.48e-03 3.21e-02f  4
   6  2.3427560e+02 4.44e+00 6.61e+02  -1.0 8.80e+00   3.4 1.59e-03 1.48e-02f  2
   7  2.3419800e+02 4.43e+00 6.49e+02  -1.0 2.06e+01   2.9 2.25e-02 2.03e-03f  3
   8  2.3403306e+02 4.41e+00 5.86e+02  -1.0 2.83e+00   3.4 2.04e-01 2.16e-02f  3
   9  2.3336538e+02 4.37e+00 4.37e+02  -1.0 1.81e+00   3.8 2.10e-01 1.84e-02f  4
iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls
  10  2.0908791e+02 1.82e+00 2.22e+04  -1.0 4.75e-01   4.2 1.00e+00 1.66e-01f  2



...

iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls
 170  1.0616914e+01 0.00e+00 1.19e-03  -5.7 8.02e-06    -  1.00e+00 2.50e-01h  3
 171  1.0616914e+01 0.00e+00 5.56e-04  -5.7 7.09e-06    -  1.00e+00 3.12e-02h  6
 172  1.0616914e+01 0.00e+00 2.51e-04  -5.7 1.19e-05    -  1.00e+00 1.00e+00h  1
 173  1.0616914e+01 0.00e+00 3.71e-04  -5.7 3.86e-06    -  1.00e+00 1.00e+00H  1
 174  1.0616914e+01 0.00e+00 4.39e-03  -5.7 3.21e-06    -  1.00e+00 1.00e+00h  1
 175  1.0616914e+01 0.00e+00 1.11e-03  -5.7 3.13e-06    -  1.00e+00 2.50e-01h  3
 176  1.0616914e+01 0.00e+00 1.38e-04  -5.7 2.23e-06    -  1.00e+00 1.00e+00H  1
 177  1.0616914e+01 0.00e+00 3.36e-04  -5.7 1.16e-05    -  1.00e+00 1.00e+00F  1
 178  1.0616914e+01 0.00e+00 1.47e-03  -5.7 6.31e-06    -  1.00e+00 1.00e+00H  1
 179  1.0616914e+01 0.00e+00 8.32e-04  -5.7 4.29e-06    -  1.00e+00 1.00e+00H  1
iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls
 180  1.0616914e+01 0.00e+00 1.63e-03  -5.7 9.86e-06    -  1.00e+00 1.00e+00H  1
 181  1.0616914e+01 0.00e+00 2.67e-03  -5.7 1.25e-05    -  1.00e+00 1.00e+00F  1
 182  1.0616914e+01 0.00e+00 2.74e-03  -5.7 2.18e-06    -  1.00e+00 1.00e+00h  1
 183  1.0616914e+01 0.00e+00 2.87e-03  -5.7 2.04e-06    -  1.00e+00 2.50e-01h  3
 184  1.0616914e+01 0.00e+00 1.41e-03  -5.7 6.77e-06    -  1.00e+00 2.50e-01h  3
 185  1.0616914e+01 0.00e+00 7.26e-04  -5.7 1.79e-06    -  1.00e+00 1.00e+00H  1
 186  1.0616914e+01 0.00e+00 3.33e-04  -5.7 5.29e-06    -  1.00e+00 2.38e-07h 23
 187r 1.0616914e+01 0.00e+00 3.76e+02  -5.7 0.00e+00    -  0.00e+00 4.77e-07R 22
 188r 9.4492203e+01 0.00e+00 3.76e+00  -5.7 3.20e+00    -  9.90e-01 1.00e+00f  1
^Crhs failed: KeyboardInterrupt
At t = 0, the right-hand side routine failed in an unrecoverable manner.
CasADi - 2020-09-03 10:20:06 WARNING("solver:nlp_jac_g failed:Error in Function::operator() for 'nlp_jac_g' [MXFunction] at .../casadi/core/function.cpp:1368:
Error in Function::operator() for 'adj1_F' [MXFunction] at .../casadi/core/function.cpp:1368:
Error in Function::operator() for 'asens1_F' [CvodesInterface] at .../casadi/core/function.cpp:1368:
.../casadi/interfaces/sundials/cvodes_interface.cpp:380: CVodeF returned "CV_RHSFUNC_FAIL". Consult CVODES documentation.") [.../casadi/core/oracle_function.cpp:223]
Sorry, things failed ?!?!```

**Comment by jgillis** [comment]
The log insists that `Ctr+C` was pressed, interrupting the process?

**Comment by juliuskittler** [comment]
> insists

Yes, I pressed `Ctrl+C`. At iteration 188, nothing happened anymore. I was waiting for several minutes, without getting any warning or error message. It was only when I pressed  `Ctrl+C` that I also got the error message printed out.

**Comment by jgillis** [comment]
Right, so it just means that cvodes (an adaptive step size solver) is suddenly taking an awful lot of steps.
You might play with the 'max_num_steps' (default 10000) to match your patience.
Hopefully, when the limit is exceeded, IPOPTs backtracking line-search will find a point in decision space that does not cause a blow-up in integration steps

**Comment by jgillis** [comment]
Also, it seems the solver is converging to 1.0616914e+01 but then suddenly starts doing a lot of back-tracking (23 iterations),
indicating that the linearized model badly matches the real model.
This may be caused by a finite-differentation subblock in your graph, or by a too-low tolerance on cvodes (try to decrease abstol (1e-8), reltol (1e-6 )). If you are lucky, you avoid going in to restoration phase (suffix r to the iteration number), and avoid that region of decision space that caused cvodes to try excessive number of steps in the first place.

**Comment by jgillis** [comment]
Indeed,  I am referring solely to integrator options in the above

**Comment by juliuskittler** [comment]
Thanks again, I got it now. Reducing `max_num_steps` in the `opts` of `casadi.integrator` prevents the `nlpsol` from getting stuck. 

By the way, is there a way to see the number of integration steps for each iteration in the printout (as another column in the printouts that I posted along with the error message)? It would be helpful so that I can set a value of `max_num_steps` that makes sense.
